### PR TITLE
`dependencies`: Upgrade sqlvirtualmachine to 2021-11-01-preview

### PR DESCRIPTION
**Pull request by myc2h6o**
To prepare for supporting `instance_settings` mentioned in #15824

### CODE REVIEW

File: internal/services/mssql/mssql_virtual_machine_resource.go [hunk 1]
```diff
@@ -811,7 +811,7 @@ func flattenSqlVirtualMachineStorageConfigurationSettings(input *sqlvirtualmachi
 		"disk_type":             string(input.DiskConfigurationType),
 		"data_settings":         flattenSqlVirtualMachineStorageSettings(input.SQLDataSettings),
 		"log_settings":          flattenSqlVirtualMachineStorageSettings(input.SQLLogSettings),
-		"temp_db_settings":      flattenSqlVirtualMachineStorageSettings(input.SQLTempDbSettings),
+		"temp_db_settings":      flattenSqlVirtualMachineTempDbSettings(input.SQLTempDbSettings),
```
Comment (myc2h6o) [inline internal/services/mssql/mssql_virtual_machine_resource.go:814]: The type has been changed in the new api version so created a copy of the old expand/flatten

**Comment by myc2h6o** [comment]
Hi @katbyte there seems to be multiple features waiting for the api upgrade, would you be able to take a look?

**Review by katbyte — CHANGES_REQUESTED** [review]
@myc2h6o - all tests are failing with this PR:

image

**Comment by myc2h6o** [comment]
Hi @katbyte I found these tests failing on main due to dangling resources during delete phase, fixing them with #16519. Please take a look.

**Comment by myc2h6o** [comment]
The fixing pr #16519 is merged, updated test result with this change:
image

Ampng these, the test `TestAccMsSqlVirtualMachine_autoPatching` is intermittently failing ever since it's introduced due to the refresh latency, similar to the other property `auto_backup`, we may need to apply the same workaround for it, will address it in a separate pr:
https://github.com/hashicorp/terraform-provider-azurerm/blob/cdd2a577a0ed4717318d575d2b03330ca68a07a8/internal/services/mssql/mssql_virtual_machine_resource.go#L397-L407

**Review by katbyte — APPROVED** [review]
LGTM 🚜

**Pull request merged**